src/runtime_src/core/common/scheduler.cpp

```diff
@@ -165,6 +165,20 @@ init(xclDeviceHandle handle, const axlf* top)
   while (ecmd->state < ERT_CMD_STATE_COMPLETED)
     while (xclExecWait(handle,1000)==0) ;
 
+  /*
+   * We ignore any error here and continue to finish. But if we have 2+
+   * applications calling init simultaneously, the second init would fail due
+   * to the ERT_CONFIGURE cmd won't not be processed during exec->active,
+   * (configure is in progress before exec->configured is set), status.
+   *
+   * should we need this?
+   * if (ecmd->state != ERT_CMD_STATE_COMPLETED)
+   *    throw std::runtime_error("xclExecBuf incompleted with state: %d", ecmd->state);
+   *
+   * Note: this would not be an issue for other boards except for versal,
```

Comment:
this is probably not exactly true. put pdi download aside, for versal or u30, any hybrid platform that use soft kernel may have this issue too. (real soft kernel, not pdi download using soft kernel config command.

Reply:
thanks, added more comments, hope that is not too much :).